### PR TITLE
Bump interfaces to ID3D12Device9 and ID3D12GraphicsCommandList6

### PR DESCRIPTION
**Pull request by doitsujin**
Lots of boilerplate here since D3D12_RESOURCE_DESC1 was introduced for sampler feedback.

### CODE REVIEW

File: include/vkd3d_d3d12.idl [hunk 1]
```diff
@@ -3313,6 +3322,44 @@ typedef struct D3D12_DISPATCH_MESH_ARGUMENTS
     UINT ThreadGroupCountZ;
 } D3D12_DISPATCH_MESH_ARGUMENTS;
 
+typedef enum D3D12_SHADER_CACHE_MODE {
```
Comment (HansKristian-Work) [inline include/vkd3d_d3d12.idl:3325]: Brace

File: libs/vkd3d/resource.c [hunk 2]
```diff
@@ -1543,17 +1544,26 @@ static void STDMETHODCALLTYPE d3d12_resource_Unmap(d3d12_resource_iface *iface,
 static D3D12_RESOURCE_DESC * STDMETHODCALLTYPE d3d12_resource_GetDesc(d3d12_resource_iface *iface,
         D3D12_RESOURCE_DESC *resource_desc)
 {
-    struct d3d12_resource *resource = impl_from_ID3D12Resource1(iface);
+    struct d3d12_resource *resource = impl_from_ID3D12Resource2(iface);
 
     TRACE("iface %p, resource_desc %p.\n", iface, resource_desc);
 
-    *resource_desc = resource->desc;
+    resource_desc->Dimension = resource->desc.Dimension;
```
Comment (HansKristian-Work) [inline libs/vkd3d/resource.c:1551]: Not related to the patch, but I noticed that this function signature looks completely different from https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12resource-getdesc. Shouldn't we return the struct by value?

File: libs/vkd3d/device.c [hunk 3]
```diff
@@ -4849,7 +4725,242 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateProtectedResourceSession1(d3
     return E_NOTIMPL;
 }
 
-CONST_VTBL struct ID3D12Device7Vtbl d3d12_device_vtbl =
+static D3D12_RESOURCE_ALLOCATION_INFO* STDMETHODCALLTYPE d3d12_device_GetResourceAllocationInfo2(d3d12_device_iface *iface,
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:4729]: Odd that we're returning a pointer. Is this some COM ABI shenanigans?

**Review by HansKristian-Work — COMMENTED** [review]
Some nits/questions. CI build has a bunch of warnings as well with pointer types that were missed.

File: libs/vkd3d/resource.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d/resource.c:1551]: This signature is correct, any function that returns a struct is turned into a function that takes a pointer to an out-struct and returns the same pointer. Don't ask me why, but this is done for all COM functions of this kind.

File: libs/vkd3d/device.c (hunk 3 above)
Comment (doitsujin) [reply libs/vkd3d/device.c:4729]: See above.

File: include/vkd3d_d3d12.idl (hunk 1 above)
Comment (doitsujin) [reply include/vkd3d_d3d12.idl]: Fixed.

File: libs/vkd3d/resource.c (hunk 2 above)
Comment (HansKristian-Work) [reply libs/vkd3d/resource.c:1551]: OK.

**Comment by HansKristian-Work** [comment]
Still a bunch of warnings: https://github.com/HansKristian-Work/vkd3d-proton/actions/runs/1481011923

**Review by HansKristian-Work — APPROVED** [review]
LGTM, thanks.

**Pull request merged**